installer/pkg/config/parser.go

```diff
@@ -42,6 +36,126 @@ func ParseConfig(data []byte) (*Cluster, error) {
 	return &cluster, nil
 }
 
+func parseInstallConfig(data []byte, cluster *Cluster) (err error) {
+	installConfig := &types.InstallConfig{}
+	err = yaml.Unmarshal(data, &installConfig)
+	if err != nil {
+		return err
+	}
+
+	cluster.Name = installConfig.Name
+	cluster.Internal.ClusterID = installConfig.ClusterID
+	cluster.Admin = Admin{
+		Email:    installConfig.Admin.Email,
+		Password: installConfig.Admin.Password,
+		SSHKey:   installConfig.Admin.SSHKey,
+	}
+	cluster.BaseDomain = installConfig.BaseDomain
+	cluster.Networking = Networking{
+		Type:        tectonicnetwork.NetworkType(installConfig.Networking.Type),
+		ServiceCIDR: installConfig.Networking.ServiceCIDR.String(),
+		PodCIDR:     installConfig.Networking.PodCIDR.String(),
+	}
+
+	for _, machinePool := range installConfig.Machines {
+		nodePool := NodePool{
+			Name: machinePool.Name,
+		}
+		if machinePool.Replicas == nil {
+			nodePool.Count = 1
+		} else {
+			nodePool.Count = int(*machinePool.Replicas)
+		}
+		cluster.NodePools = append(cluster.NodePools, nodePool)
+		switch machinePool.Name {
+		case "master":
+			if machinePool.PlatformConfig.AWS != nil {
+				cluster.AWS.Master = aws.Master{
+					EC2Type:     machinePool.PlatformConfig.AWS.InstanceType,
+					IAMRoleName: machinePool.PlatformConfig.AWS.IAMRoleName,
+					MasterRootVolume: aws.MasterRootVolume{
+						IOPS: machinePool.PlatformConfig.AWS.EC2RootVolume.IOPS,
+						Size: machinePool.PlatformConfig.AWS.EC2RootVolume.Size,
+						Type: machinePool.PlatformConfig.AWS.EC2RootVolume.Type,
+					},
+				}
+			}
+		case "worker":
+			if machinePool.PlatformConfig.AWS != nil {
+				cluster.AWS.Worker = aws.Worker{
+					EC2Type:     machinePool.PlatformConfig.AWS.InstanceType,
+					IAMRoleName: machinePool.PlatformConfig.AWS.IAMRoleName,
+					WorkerRootVolume: aws.WorkerRootVolume{
+						IOPS: machinePool.PlatformConfig.AWS.EC2RootVolume.IOPS,
+						Size: machinePool.PlatformConfig.AWS.EC2RootVolume.Size,
+						Type: machinePool.PlatformConfig.AWS.EC2RootVolume.Type,
+					},
+				}
+			}
+		default:
+			return fmt.Errorf("unrecognized machine pool %q", machinePool.Name)
```

Comment:
These names feel more like types. Additionally it looks like there's not a lot of validation here, do we require two pools with these exact names? What should happen if we see multiple worker or master pool names?

Maybe too preliminary for that but just wanted to mention.

Reply:
> These names feel more like types.

I agree.  @abhinavdahiya, can I replace:

```go
// Machines is the list of MachinePools that need to be installed.
Machines []MachinePool `json:"machines"`
```

with:

```go
Master MachinePool `json:"master"

Worker MachinePool `json:"worker"
```

or similar?

Reply:
This would leave us with a little issue in Hive. If we use InstallConfig directly in our ClusterDeployment, this would be the long term definition of all machine pools. If we go to just one master, and one worker, then we would need to duplicate this info higher up in our own portion of ClusterDeployment for the long term definition of what machine pools should exist. That would mean a little duplication of data, and a strange section of install config that was only used during install.

Kinda feels like this should be capable of being *the* canonical source of machine sets that should exist in the cluster.

Reply:
@wking I think we decided that machinepools remain as is for now. no changes. we can expand them if needed later.

Reply:
> If we go to just one master, and one worker...

Just to be clear, I was suggesting one master *pool* (and one worker pool).

> I think we decided that machinepools remain as is for now. no changes

So just die screaming if `master` (say) is listed multiple times or not listed at all?  And silently ignore entries with names that aren't `master` or `worker`?

Reply:
I would suggest leaving name a freeform field, add an explicit type, technically the two types should probably be "control-plane" and "compute", verify you have exactly one control-plane, and at least one compute pool.